main.go

```diff
@@ -70,86 +73,84 @@ func main() {
 		go sl.Stream(logstream)
 	}
 
+	fluentServer, err := net.Listen("tcp", cfg.ListenAddress)
+	if err != nil {
+		log.Fatalf("Unable to listen: %s", err)
+	}
+
 	for {
-		containers, err := listContainers()
+		conn, err := fluentServer.Accept()
 		if err != nil {
-			log.Fatalf("Unable to list containers: %s", err)
+			log.Fatalf("Unable to accept tcp connection: %s", err)
 		}
 
-		touched := []string{}
-		for _, container := range containers {
-			touched = append(touched, container.ID)
-
-			// If we don't have a channel for this container create it
-			if _, ok := receiveChannels[container.ID]; !ok {
-				log.Printf("INFO: Found container %s, attaching...", container.ID)
-				receiveChannelsLock.Lock()
-				receiveChannels[container.ID] = &bufferedLogMessageWriter{
-					channel:   logstream,
-					container: container,
-					buffer:    []byte{},
+		go handleFluentdForwardConnection(conn)
+	}
+}
+
+func handleFluentdForwardConnection(c net.Conn) {
+	defer c.Close()
+	buffer := bytes.NewBuffer([]byte{})
+
+	for {
+		tmp := make([]byte, 256)
```

Comment:
Not sure if this has any performance impact, but this could be allocated once per connection. Maybe even on the stack?